hbase-client/src/main/java/org/apache/hadoop/hbase/client/replication/ReplicationPeerConfigUtil.java

```diff
@@ -455,36 +458,45 @@ public static ReplicationPeerConfig appendTableCFsToReplicationPeerConfig(
 
   /**
    * Helper method to add base peer configs from Configuration to ReplicationPeerConfig
-   * if not present in latter.
    *
    * This merges the user supplied peer configuration
    * {@link org.apache.hadoop.hbase.replication.ReplicationPeerConfig} with peer configs
    * provided as property hbase.replication.peer.base.configs in hbase configuration.
-   * Expected format for this hbase configuration is "k1=v1;k2=v2,v2_1". Original value
-   * of conf is retained if already present in ReplicationPeerConfig.
+   * Expected format for this hbase configuration is "k1=v1;k2=v2,v2_1".
    *
    * @param conf Configuration
    * @return ReplicationPeerConfig containing updated configs.
    */
-  public static ReplicationPeerConfig addBasePeerConfigsIfNotPresent(Configuration conf,
+  public static ReplicationPeerConfig updateReplicationBasePeerConfigs(Configuration conf,
     ReplicationPeerConfig receivedPeerConfig) {
-    String basePeerConfigs = conf.get(HBASE_REPLICATION_PEER_BASE_CONFIG, "");
+    String removeBasePeerConfigs = conf.get(HBASE_REPLICATION_PEER_REMOVE_BASE_CONFIG, "");
     ReplicationPeerConfigBuilder copiedPeerConfigBuilder = ReplicationPeerConfig.
       newBuilder(receivedPeerConfig);
-    Map<String,String> receivedPeerConfigMap = receivedPeerConfig.getConfiguration();
 
+    // remove the peer configurations specified in the conf
+    if (removeBasePeerConfigs.length() != 0) {
+      List<String> removeBasePeerConfigList = Splitter.on(';').trimResults()
+        .omitEmptyStrings().splitToList(removeBasePeerConfigs);
+      for (String peerConfigToRemove : removeBasePeerConfigList) {
+        copiedPeerConfigBuilder.removeConfiguration(peerConfigToRemove);
+      }
+    }
+
+    Map<String, String> receivedPeerConfigMap = receivedPeerConfig.getConfiguration();
+    String basePeerConfigs = conf.get(HBASE_REPLICATION_PEER_BASE_CONFIG, "");
```

Comment:
What happens if I try to add a peer config setting and remove the same peer config setting in the same Configuration object? Looks like the add just silently wins? I'd think that should produce an exception (or at the very least a log).

Reply:
@gjacoby126 That is correct. I will add a log here.